src/gpuarray/buffer.h

```diff
@@ -757,6 +763,48 @@ typedef enum _ga_usefl {
   GA_USE_OPENCL =   0x4000,
 } ga_usefl;
 
+/************************************************************************************
+*                         multi-gpu communicator interface                         *
+************************************************************************************/
+
+/**
+ * \brief Creates a new communicator (multi process version)
+ * \param clique_id [const char*] TODO
+ * \param ndev [int] TODO
+ * \param rank [int] TODO
+ * \param res [int*] TODO
+ * \return gpucomm* TODO
+ */
+GPUARRAY_PUBLIC gpucomm* gpucomm_init(const char* clique_id, int ndev, int rank,
+                                      int* res);
+
+/**
+ * \brief Frees resources associated with communicator object.
+ * \param comm [gpucomm*] TODO
+ * \return void TODO
+ */
+GPUARRAY_PUBLIC void gpucomm_destroy(gpucomm* comm);
+
+/**
+ * \brief Returns nice error message.
+ * \param comm [gpucomm*] TODO
+ * \param err [int] TODO
+ * \return const char* TODO
+ */
+GPUARRAY_PUBLIC const char* gpucomm_error(gpucomm* comm, int err);
+
+// TODO cuda extension:
+// 1. ncclGetUniqueId
+// 2. ncclCommCount (below mby ops in gpucomm struct?!)
+// 3. ncclCommUserRank
+// 4. ncclCommCuDevice
```

Comment:
The unique clique id needs to be exposed in order to support multi-node/gpu as every call to `ncclGetUniqueId` returns a different id every time that needs to be shared among participants who will compose a CommWorld.

Quoting [this](https://devblogs.nvidia.com/parallelforall/fast-multi-gpu-collectives-nccl/): 
"This function assumes that the GPU belonging to the specified rank has already been selected using cudaSetDevice(). nGPUs is the number of GPUs in the clique. cliqueId allows the ranks of the clique to find each other. The same cliqueId must be used by all ranks. To achieve this, call ncclGetUniqueId() in one rank and broadcast the resulting uniqueId to the other ranks of the clique using the communication framework of your choice (for example, MPI_Bcast)."

And [this](https://github.com/NVIDIA/nccl/blob/master/test/mpi/mpi_test.cu#L62) test example of theirs.